datastore/db/metadata/inspection/__init__.py

```diff
@@ -0,0 +1,300 @@
+"""
+This module contains the function to generate the metadata necessary to interact with the database and the other layers of Fertiscan for all the inspection related objects. 
+The metadata is generated in a json format and is used to store the metadata in the database.
+
+"""
+
+from typing import List
+from pydantic import BaseModel, ValidationError
+from typing import Optional
+
+class MissingKeyError(Exception):
+    pass
+
+class NPKError(Exception):
+    pass
+
+class MetadataFormattingError(Exception):
+    pass
+
+class OrganizationInformation(BaseModel):
+    name: str
+    address: str
+    website: str
+    phone_number: str
+    
+class Value(BaseModel):
+    value: Optional[float] = None
+    unit: Optional[str] = None
+    name: str
+
+class ValuesObjects(BaseModel):
+    en: List[Value]
+    fr: List[Value]
+
+class SubLabel(BaseModel):
+    en: List[str]
+    fr: List[str]
+    
+class Metric(BaseModel):
+    value: float
+    unit: str
+
+class Metrics(BaseModel):
+    weight: List[Metric]
+    volume: Metric
+    density: Metric
+
+class ProductInformation(BaseModel):
+    name: str
+    registration_number: str
+    lot_number: str
+    metrics: Metrics
+    npk: str
+    warranty: str
+    n: float
+    p: float
+    k: float
+
+
+class Specification(BaseModel):
+    humidity: float
+    ph: float
+    solubility: float
+
+class Specifications(BaseModel):
+    en: List[Specification]
+    fr: List[Specification]
+
+class Inspection(BaseModel):
+    company: OrganizationInformation
+    manufacturer: OrganizationInformation
+    product: ProductInformation
+    cautions: SubLabel
+    instructions: SubLabel
+    micronutrients: ValuesObjects
+    ingredients: ValuesObjects
+    specifications: Specifications
+    first_aid: SubLabel
+    guaranteed_analysis: List[Value]
+
+
+def build_inspection_import(analysis_form: dict) -> str:
+    """
+    This funtion build an inspection json object from the pipeline of digitalization analysis.
+    This serves as the metadata for the inspection object in the database.
+
+    Parameters:
+    - analysis_form: (dict) The digitalization of the label.
+
+    Returns:
+    - The inspection db object in a string format.
+    """
+    try:
+        requiered_keys = [
```

Comment:
Pydantic could help with this validation if `analysis_form` was a model.

Reply:
Explaining why we didnt use the Pydantic for the analysis_form: This is the only time and occurence we will have to deal with the form with this key format. Creating the Pydantic model for the analysis_form would also requiered to write alot of code for a structure that is already validated in the BE.